packages/node_modules/samples/browser-auth-implicit/test/wdio/spec/implicit-grant-authentication.js

```diff
@@ -23,15 +25,33 @@ describe(`samples/browser-auth-implicit`, () => {
     });
 
     step(`enters user login information`, () => {
-      // Enter Email
-      browserSpock.waitForExist(`input[type=email]`);
-      browserSpock.setValue(`[placeholder="Email address"]`, spock.email);
-      browserSpock.click(`[name="btnOK"]`);
-
-      // Enter Password
-      browserSpock.waitForExist(`input[type=password]`);
-      browserSpock.setValue(`[placeholder="Password "]`, spock.password);
-      browserSpock.click(`[name="Login.Submit"]`);
+      // the login pages aren't known for their consistency, so we'll break with
+      // out pattern of using accessibility labels for test selectors in favor
+      // of using the well known (if not a11y helpful) ids.
+      browserSpock.waitForExist(`#IDToken1`);
+
+      // Enter Email if we haven't already filled it in
+      if (!browserSpock.isVisible(`#IDToken1[readonly]`)) {
+        browserSpock.setValue(`#IDToken1`, spock.email);
+        browserSpock.click(`#IDButton2`);
+      }
+      for (let i = 0; i < 3; i++) {
+        console.log(`password entry attempt #${i}`);
+        browserSpock.waitForExist(`#IDToken2`);
+
+
+        browserSpock.setValue(`#IDToken2`, spock.password);
+        browserSpock.click(`#Button1`);
+
+        if (!browserSpock.isVisible(`.generic-error`)) {
+          break;
+        }
+      }
+
+      // I feel like there's a better way to do this...
```

Comment:
Could you check to see if you got redirected after a successful login? if the `.generic-error` class is on the redirected page, it could cause a false negative

Reply:
if '.generic-error' is on the redirected page, we want it to fail.